Updated tso pipeline to add in cnv caller logic

alexiswl opened the pull request:
Added extra wgs metrics to collect from AlignCollapseFusionCaller step.

Added cnv caller bins and wrap into a compressed json

Added replay json to outputs

pdiakumis commented:
@alexiswl sorry, one more, can we also grab the non-json `fragment_length_hist.csv` to be consistent with the other Dragen workflows? e.g. 

```
gds://production/temp/analysis_data/SBJ04420/tso_ctdna_tumor_only/20231124b9c9d854/L2301406/steps/run_tso_ctdna_workflow_step/0/steps/run_tso500_ctdna_analysis_workflow_step/try-1/analysis_workflow/analysis_outputs/AlignCollapseFusionCaller/PRJ230810_L2301406/PRJ230810_L2301406.fragment_length_hist.csv
```

alexiswl commented:
> one more, can we also grab the non-json fragment_length_hist.csv to be consistent with the other Dragen workflows? e.g.

Resolved in e2eecccb056e38e13dc2e762d98e59b3f0ee7b49

See outputs with

```
gds-ls gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/
```

Gives

```
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/evidence.PRJ210166_L2101400.bam
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/evidence.PRJ210166_L2101400.bam.bai
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.AlignCollapseFusionCaller_metrics.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.bam
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.bam.bai
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.bam.md5sum
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.cleaned.stitched.bam
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.cleaned.stitched.bam.bai
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.CnvCallerBinCounts.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_CombinedVariantOutput.tsv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_CopyNumberVariants.vcf.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_CopyNumberVariants.vcf.gz.tbi
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_ctDNA_alignment_collapse_fusion_caller_multiqc/
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_Failed_Exon_coverage_QC.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_Failed_Exon_coverage_QC.txt
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.fastqc_metrics.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.fragment_length_hist.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.fragment_length_hist.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_Fusions.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_Fusions.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.mapping_metrics.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_MergedSmallVariantsAnnotated.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_MergedSmallVariantsAnnotated.json.gz.jsi
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_MergedSmallVariants.genome.vcf.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_MergedSmallVariants.genome.vcf.gz.tbi
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_MergedSmallVariants.vcf.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_MergedSmallVariants.vcf.gz.tbi
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.msi.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_normalizedBinCount.tsv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_rawBinCount.tsv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400-replay.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_SampleAnalysisResults.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.sv_metrics.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.TargetRegionCoverage.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.time_metrics.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.tmb.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_TMB_Trace.json.gz
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400_TMB_Trace.tsv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.trimmer_metrics.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.umi_metrics.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.wgs_contig_mean_cov.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.wgs_coverage_metrics.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.wgs_fine_hist.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.wgs_hist.csv
gds://development/temp/alexis/cttso-cnv-bin-test/output/20231219_141001/Results/PRJ210166_L2101400/PRJ210166_L2101400.wgs_overall_mean_cov.csv
```

pdiakumis reviewed (approved):
Thanks heaps, LGTM 👍
